### PR TITLE
Family rotation

### PR DESCRIPTION
**Pull request by ShengxiZhang**
Add a new node to set the rotation of the in-placed family instances.

### CODE REVIEW

**Comment by ShengxiZhang** [comment]
@LianweiHuang 
@XiaojieZhang
@aandu

**Comment by ShengxiZhang** [comment]
@kiki-zhang

**Comment by ShengxiZhang** [comment]
@ikeough  @Randy-Ma

File: src/Libraries/RevitNodes/Elements/FamilyInstance.cs [hunk 1]
```diff
@@ -298,5 +300,39 @@ public override string ToString()
...
+           double[] oldRotationAngles;
+           TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
+
+           Double newRotationAngle = degree * Math.PI / 180;
+
+           if (!oldRotationAngles[0].AlmostEquals(newRotationAngle, 1.0e-6))
+           {
+              double rotateAngle = newRotationAngle - oldRotationAngles[0];
+              var axis = Line.CreateUnbound(oldTransform.Origin, oldTransform.BasisZ);
+              ElementTransformUtils.RotateElement(Document, new ElementId(Id), axis, -rotateAngle);
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: Where does this function come from?

File: src/Libraries/RevitNodes/Elements/InternalUtilities/TransformUtils.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+using Autodesk.Revit.DB;
+using System.Windows.Media.Media3D;
+
+namespace Revit.Elements.InternalUtilities
+{
+   [IsVisibleInDynamoLibrary(false)]
+   public class TransformUtils
+   {
+      // decomp_ZYX
+      public static void ExtractEularAnglesFromTransform(Transform transform, out double[] angles )
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/InternalUtilities/TransformUtils.cs]: ExtractEulerAnglesFromTransform?

Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/InternalUtilities/TransformUtils.cs]: Oh! Sorry for the typo! Eular=>Euler

**Comment by Randy-Ma** [comment]
@ShengxiZhang 
Have you tried running a graph with the new node? I doubt there will be an infinite loop.

File: src/Libraries/RevitNodes/Elements/FamilyInstance.cs (hunk 1 above)
Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: This is from RevitAPI assembly.

**Comment by ShengxiZhang** [comment]
Yes, I tried the node. There is no infinite loop because here we set a fixed rotation, so from the 2nd time, the document won't be changed so the node won't be executed.

**Comment by Randy-Ma** [comment]
There should be no infinite loops some time back. But now it is slightly different. Could you help add a system test case? In the test case, open and run a .dyn file instead of calling APIs.

File: src/Libraries/RevitNodes/Elements/FamilyInstance.cs (hunk 1 above)
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: Thanks!
But I do not understand why we put a minus sign before rotateAngle.

Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: Ah, from the testing, it looks the rotation direction is different between the euler angle and RotateElement.

Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: But then if the rotating is not along a vector, how shall we define the rotation?

Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/FamilyInstance.cs]: We define the rotation is always along its local Z-axis, this is why we removed the argument rotation_axis. Does it make sense?

**Comment by ShengxiZhang** [comment]
@Randy-Ma 
Hi Randy, I have added the system test case, could you please review it? Thank you!

**Comment by Randy-Ma** [comment]
@ShengxiZhang 
Great! Thanks a lot! LGTM!

**Comment by Randy-Ma** [comment]
But do not forget to cross-merge the changes here.

**Pull request merged**